### PR TITLE
[next] Report Comment + Stream Refactor & Test Improvements

### PR DESCRIPTION
**Pull request by alinhle**
## What does this PR do?
- https://www.pivotaltracker.com/story/show/159770747
- https://www.pivotaltracker.com/story/show/162791344
- Implements Report Comment feature
- Fix a bug with removing Reaction
- Refactored Permalink
- Improved all Stream feature tests and removed a ton of snapshots
- Added integration test for Reaction
- Added integration test for Report Comment

## How do I test this PR?

- Report a comment

### CODE REVIEW

File: src/core/client/stream/tabs/comments/containers/ReportButtonContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  comment: CommentData;
+  me: MeData | null;
+  showAuthPopup: ShowAuthPopupMutation;
+}
+
+class ReportButtonContainer extends React.Component<
+  ReportButtonContainerProps
+> {
+  private get loggedIn() {
+    return this.props.me;
```
Comment (wyattjoh) [inline src/core/client/stream/tabs/comments/containers/ReportButtonContainer.tsx:28]: It seems like this is used frequently.. Is there some way to abstract this? Even adding a note that it should be extracted would be fine.

File: src/core/client/stream/tabs/comments/views/reportComment/components/ReportCommentForm.tsx [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+import { validateMaxLength } from "talk-framework/lib/validation";
+import {
+  Button,
+  Flex,
+  HorizontalGutter,
+  RadioButton,
+  Typography,
+} from "talk-ui/components";
+
+import { PropTypesOf } from "talk-ui/types";
```
Comment (wyattjoh) [inline src/core/client/stream/tabs/comments/views/reportComment/components/ReportCommentForm.tsx]: Review import ordering, a bit off here and there.. Is there a tslint linting package we can use to help with this??

File: src/core/client/stream/tabs/comments/views/reportComment/containers/ReportCommentFormContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        });
+      }
+
+      this.setState({ done: true });
+    } catch (error) {
+      if (error instanceof BadUserInputError) {
+        return error.invalidArgsLocalized;
+      }
+      // tslint:disable-next-line:no-console
+      console.error(error);
```
Comment (wyattjoh) [inline src/core/client/stream/tabs/comments/views/reportComment/containers/ReportCommentFormContainer.tsx:59]: We should keep note of these in the future to attach some style of client side error reporting.

File: src/locales/en-US/stream.ftl [hunk 4]
```diff
@@ -85,6 +85,9 @@ comments-permalinkView-singleConversation = SINGLE CONVERSATION
 comments-inReplyTo = In reply to <username></username>
 comments-replyTo = Replying to: <username></username>
 
+comments-reportButton-report = Report
+comments-reportButton-reported = Reported
+
 ## Profile Tab
 profile-historyComment-viewConversation = View Conversation
 profile-historyComment-replies = Replies {$replyCount}
```
Comment (wyattjoh) [inline src/locales/en-US/stream.ftl:93]: Obviously unrelated to this PR, but could you fix this so when there is one reply, it doesn't read as `Replies 1`?

File: src/core/client/stream/tabs/comments/views/reportComment/containers/ReportCommentFormContainer.tsx (hunk 3 above)
Comment (alinhle) [reply src/core/client/stream/tabs/comments/views/reportComment/containers/ReportCommentFormContainer.tsx:59]: VS Code -> find -> console.error :-)

File: src/core/client/stream/tabs/comments/containers/ReportButtonContainer.tsx (hunk 1 above)
Comment (alinhle) [reply src/core/client/stream/tabs/comments/containers/ReportButtonContainer.tsx:28]: I don't think there is something to be extracted here. How would that look like?

File: src/core/client/stream/tabs/comments/views/reportComment/components/ReportCommentForm.tsx (hunk 2 above)
Comment (alinhle) [reply src/core/client/stream/tabs/comments/views/reportComment/components/ReportCommentForm.tsx]: Couldn't find one unfortunately :-(

**Review by wyattjoh — APPROVED** [review]

**Pull request merged**